webapp/src/Controller/API/JudgehostController.php

```diff
@@ -614,8 +614,16 @@ public function updateJudgingAction(Request $request, string $hostname, int $jud
                     $this->dj->auditlog('judging', $judgingId, 'judged',
                                         'compiler-error', $hostname, $contestId);
 
+                    if ($judging->getRejudgingid() !== null && $judging->getRejudging()->getAutoApply()) {
+                        $judging->getSubmission()->setRejudging(null);
+                        foreach ($judging->getSubmission()->getJudgings() as $j) {
+                            $j->setValid(false);
+                        }
+                        $judging->setValid(true);
+                        $this->em->flush();
```

Comment:
I think it makes sense to abstract this block of code away in a separate function called `setActiveJudging` or something like that, not sure if as a method on a submission or judging object. That also removes th same duplicate below.

Reply:
Done.